app/assets/javascripts/react_on_rails.js

```diff
@@ -156,14 +157,14 @@
     return consoleReplay;
   };
 
-  function createReactElement(componentName, propsVarName, props, domId, trace, generatorFunction) {
+  function createReactElement(componentName, propsVarName, props, domId, trace, generatorFunction, location) {
     if (trace) {
       console.log('RENDERED ' + componentName + ' with data_variable ' +
         propsVarName + ' to dom node with id: ' + domId);
     }
 
     if (generatorFunction) {
-      return this[componentName](props);
+      return this[componentName](props, location);
```

Comment:
This might return an object of shape: `{ routeRedirect }` rather than a React component.

See https://github.com/shakacode/react-webpack-rails-tutorial/pull/128/files#diff-53c7f33b8c082f3109fc40557116e492R37